src/commands/app/deploy.js

```diff
@@ -337,6 +340,11 @@ Deploy.flags = {
     description: '[default: true] Update log forwarding configuration on server',
     default: true,
     allowNo: true
+  }),
+  code: flags.boolean({
+    description: '[default: true] --no-code prevents deploying action code',
```

Comment:
- `--no-code` could be confusing for apps that have UI code too, maybe we should rename this to be more specific, what about `--no-action-code` or `--no-action-update`
- also should we add the flag to `app run` too ?
cc @purplecabbage @shazron

Reply:
I agree on the confusion factor.  I would much prefer if this was just another command.
Can we enumerate all the flag combos with all the expected operations?
```
aio app deploy --no-web-assets # actions and their params are deployed
aio app deploy --no-actions  # web-assets are deployed
aio app deploy --no-code     # web-assets and params are deployed
```
The pattern this is developing into is moving in the wrong direction imho.  I think we should be specifying what we _want_ to do, not what we don't want.

It might make more sense to leave this as just an `aio runtime` command ...

Reply:
I think specifying what we want to do instead of what we don't want is a re-factor decision. This would have been better for sure with (borrowing from Firebase) `--only functions` etc so we could have: `--only action-params` for example.

For now we have this pattern so this new flag has to fit in to that pattern. `--no-actions` means no action code and thus no params are deployed (thus for both). So to specify that action params are deployed, but *not* action code, I would say the flag is `--no-action-code`. Does deploying action code but not action params make sense? I don't think that the api allows for it anyway, or that we need to allow for it, so we won't need a flag for that scenario.

Expected to deploy:
| flag                             | actions | action params | web assets |
|----------------------------------|---------|---------------|------------|
| --no-actions                     | N       | N             | Y          |
| --no-action-code                 | N       | Y             | Y          |
| --no-web-assets                  | Y       | Y             | N          |
| --no-web-assets --no-action-code | N       | Y             | N          |

Reply:
This is the full truth table:
| flag                             | actions | action params | web assets |
|----------------------------------|---------|---------------|------------|
| --no-actions --no-web-assets     | N       | N             | N          |
| --no-actions                     | N       | N             | Y          |
| --no-action-code --no-web-assets | N       | Y             | N          |
| --no-action-code                 | N       | Y             | Y          |
| (invalid)                        | Y       | N             | N          |
| (invalid)                        | Y       | N             | Y          |
| (no flags)                       | Y       | Y             | Y          |

Reply:
Following the discussion, I have changed the name of flag from `--no-code` to `--no-action-code` to avoid confusion with the UI code as stated by @moritzraho. The behavior of this flag is intended to be precisely as @shazron mentioned in the truth table.